src/core/DataFile.cpp

```diff
@@ -1762,6 +1764,27 @@ void DataFile::upgrade_bbTcoRename()
 }
 
 
+//! Update MIDI CC indexes, so that they are counted from 0. Older releases of LMMS
+//! count the CCs from 1.
+void DataFile::upgrade_midiCCIndexing()
+{
+	std::vector<const char*> attributesToUpdate = {"inputcontroller", "outputcontroller"};
+
+	QDomNodeList elements = elementsByTagName("Midicontroller");
+	for(int i = 0; !elements.item(i).isNull(); ++i)
+	{
+		auto element = elements.item(i).toElement();
```

Comment:
```suggestion
		if (elements.item(i).isNull()) { continue; }
		auto element = elements.item(i).toElement();
```

Reply:
Reviewed too fast in the previous one, forgot to add this change.

Reply:
Added this line, but I don't really understand the point. Under which conditions _element_ may be null?

Reply:
Invalid XML or similar. If we have elements A, B, C and element B is invalid in some way which makes it null, in prior loop we would end on B and skip C entirely. With new loop, we need to check if null to continue processing and not try to process on a null element.